core/base/persistenceDiagramClustering/PDBarycenterImpl.h

```diff
@@ -687,9 +680,9 @@ typename PDBarycenter<dataType>::KDTreePair
   auto correspondance_kdt_map
     = kdt->build(coordinates.data(), barycenter_goods_[0].size(), dimension,
                  weights, barycenter_goods_.size());
-  if(debugLevel_ > 3)
-    std::cout << "[Building KD-Tree] Time elapsed : " << tm.getElapsedTime()
-              << " s." << std::endl;
+  this->printMsg("[Building KD-Tree] Time elapsed : "
```

Comment:
see https://github.com/topology-tool-kit/ttk/blob/dev/core/base/helloWorld/HelloWorld.h (l. 123) for an example of time logging